modules/templates/util.go

```diff
@@ -45,3 +49,74 @@ func dict(args ...any) (map[string]any, error) {
 	}
 	return m, nil
 }
+
+func dumpVarMarshalable(v any, dumped map[uintptr]bool) (ret any, ok bool) {
```

Comment:
Shouldn't we make `dumped` a `set[uintptr]`?

Reply:
Golang doesn't have builtin `set`, most `set` packages are using `map` internally.

Reply:
I know, but I think we've made our own set type already

Reply:
But we really do not need that .... the `map[uintptr]bool` is quite clear here.

If the `Set[T]` could save some code or improve some performance, I would agree to use, but I don't see a strong reason.

Reply:
Other code needs the `Set[T]` because they need to do `SetOf` or `Values`, but here, we only need to check ....

Reply:
If I remember correctly, our set is a `map[T]struct{}{}` which apparently is more performant…

Reply:
No

```go
func BenchmarkMapBool(b *testing.B) {
	for i := 0; i < b.N; i++ {
		m := map[int]bool{}
		for j := 0; j < 100; j++ {
			m[j] = true
		}
	}
}
func BenchmarkMapSet(b *testing.B) {
	for i := 0; i < b.N; i++ {
		m := container.Set[int]{}
		for j := 0; j < 100; j++ {
			m.Add(j)
		}
	}
}
```

```
goos: darwin
goarch: amd64
pkg: code.gitea.io/gitea/modules/templates
cpu: Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
BenchmarkMapBool
BenchmarkMapBool-12    	  216259	      5096 ns/op
BenchmarkMapSet
BenchmarkMapSet-12     	  194236	      5962 ns/op
```

Reply:
The generic functions are usually slower than the non-generic ones.

Using non-generic `struct{}{}` could be slightly faster than `bool`, but I do not like the `struct{}{}` syntax here, it makes code looks messy.

And this function is only a debug helper, it won't run in production, so we do not need to pay too much attention on its performance.

Reply:
I know, I resolved this conversation already which means I'm also fine with the current approach.